### PR TITLE
Remove junit-quickcheck from Corda because it is only used for testing.

### PR DESCRIPTION
**Pull request by chrisr3**
The `junit-quickcheck-*` artifacts are for testing only, and so do not need to be included in `corda.jar`.

### CODE REVIEW

File: client/jackson/build.gradle [hunk 1]
```diff
@@ -23,5 +23,12 @@ dependencies {
     // This adds support for java.time types.
     compile "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:$jackson_version"
 
+    testCompile project(':core').sourceSets.test.output
```
Comment (Clintonio) [inline client/jackson/build.gradle]: testCompile project(':core') should be enough?

File: client/jackson/build.gradle [hunk 2]
```diff
@@ -23,5 +23,12 @@ dependencies {
...
     testCompile "junit:junit:$junit_version"
+
+    // TODO: Upgrade to junit-quickcheck 0.8, once it is released,
+    //       because it depends on org.javassist:javassist instead
+    //       of javassist:javassist.
+    testCompile "com.pholser:junit-quickcheck-core:$quickcheck_version"
```
Comment (Clintonio) [inline client/jackson/build.gradle]: If we're depending on core these dependencies should be dragged in automatically right?

File: client/jackson/build.gradle (hunk 1 above)
Comment (chrisr3) [inline client/jackson/build.gradle]: The `:client:jackson` tests are depending on _test_ classes from `:core`. Using

    testCompile project(':core')

would only create a dependency on the _main_ classes.

File: client/jackson/build.gradle (hunk 2 above)
Comment (chrisr3) [inline client/jackson/build.gradle]: It doesn't seem to work that way for test dependencies.

File: client/jackson/build.gradle (hunk 1 above)
Comment (Clintonio) [inline client/jackson/build.gradle]: Fair. I presumed, perhaps incorrectly, that testCompile would resolve only to the test sources.

**Comment by Clintonio** [comment]
You'll want to rerun the PR by the looks of it.

**Pull request merged**